### ISSUE TITLE
Master instance update using master_instance_type = "<instance-type>"  fails the update and doesn't change the instance type of master

### ISSUE THREAD

**Issue by smrutiranjantripathy** [aws/aws-parallelcluster#1628]
**Environment:**
 - N/A
 
**Bug description and how to reproduce:**
If we have a requirement to change the master instance type and we change it by changing the parameter  "master_instance_type" in config and do a update , then the update fails with the below error"
```
Interface: [eni-*******] in use. (Service: AmazonEC2; Status Code: 400; Error Code: InvalidNetworkInterface.InUse; Request ID: 4e51a13b-e440-4eba-a2b6-d4ff40d0208e)
```

It is because, it is trying to use earlier eni provisioned for the master node.  It logs the error message in coudtrail as :
  ```
    "eventTime": "2020-01-30T21:16:51Z",
    "eventSource": "ec2.amazonaws.com",
    "eventName": "RunInstances",
    "awsRegion": "us-west-2",
    "sourceIPAddress": "cloudformation.amazonaws.com",
    "userAgent": "cloudformation.amazonaws.com",
    "errorCode": "Client.InvalidNetworkInterface.InUse",
    "errorMessage": "Interface: [eni-*********] in use.
```

The workaround at the moment is deleting the master instance manually from console and then doing `pcluster update` .  I expect that pcluster should configure cloudformation in such a way that it handles the update for master instance.

Kindly let me know if anything else is required.

**Additional context:**
Any other context about the problem. E.g.:
 - N/A

**Comment by rexcsn** [comment]
Hi,

Thank you for reporting this issue. This is a bug in v2.5.0 and v2.5.1. We are investigating on this issue and working on a fix.
In the meantime, the workaround you mentioned seems to be the best way around this problem for now.

**Comment by rexcsn** [comment]
Update on this issue. The workaround you mentioned will work in specific settings but please be aware it has several risks:
* Master instance has to be terminated and data on the instance may be lost during this process
* After terminating the master instance you need to execute `pcluster update` quickly. Once CloudFormation detects that the old master instance is terminated it will not process future update requests, and the stack/cluster will be completely unusable as there is no way to update or restore master instance.

Because of these risks, we do not recommend using this workaround.

For now, the safest way to update `master_instance_type` is to do it manually through the EC2 console. Please find and stop(not terminate) the master instance, then change the instance type from Actions -> Instance Settings -> Change Instance Type

**Comment by bollig** [comment]
Ran into this in v2.6.1 and v2.7.0 as well. Unfortunately, after the first update fails and rolls back, subsequent "pcluster update" attempts with the original config also fail. To get pcluster control of the master again I have to delete and recreate the cluster.

**Comment by bollig** [comment]
Also found this same error when I attempt to update an existing cluster to add `enable_efa = compute` .

**Comment by enrico-usai** [comment]
This issue has been addressed as part of the #1583

With this patch, the `pcluster update` command will check for the updatablity of each detected change and will prevent the update from being run if any change is marked as not safe or not possible.
